app/main.py

```diff
@@ -1,34 +1,84 @@
+from colorama import Fore, Style, init
+from typing import List
+init()
+
+
 class Deck:
-    def __init__(self, row, column, is_alive=True):
-        pass
+    def __init__(self, row: int, column: int, is_alive: bool = True) -> None:
+        self.row = row
+        self.column = column
+        self.is_alive = is_alive
 
 
 class Ship:
-    def __init__(self, start, end, is_drowned=False):
-        # Create decks and save them to a list `self.decks`
-        pass
+    def __init__(
+            self, start: tuple,
+            end: tuple,
+            is_drowned: bool = False
+    ) -> None:
+
+        self.start = start
+        self.end = end
+        self.is_drowned = is_drowned
+        self.deck = []
+
+        self.create_ship()
+
+    def create_ship(self) -> None:
+        for coord_x in range(
+                self.start[0], self.end[0] + 1
+        ):
 
-    def get_deck(self, row, column):
-        # Find the corresponding deck in the list
-        pass
+            for coord_y in range(self.start[1], self.end[1] + 1):
+                if coord_x == self.start[0] or coord_y == self.start[1]:
+                    self.deck.append(Deck(coord_x, coord_y))
 
-    def fire(self, row, column):
-        # Change the `is_alive` status of the deck
-        # And update the `is_drowned` value if it's needed
-        pass
+    def get_deck(self, row: tuple, column: tuple) -> tuple:
+        for deck in self.deck:
+            if deck.row == row and deck.column == column:
+                return deck
+
+    def fire(self, row: tuple, column: tuple) -> None:
+        deck = self.get_deck(row, column)
+        if deck is not None:
+            deck.is_alive = False
+            if all(not d.is_alive for d in self.deck):
+                self.is_drowned = True
```

Comment:
```suggestion
            self.is_drowned = all(not d.is_alive for d in self.deck):
```
Also rename d to deck